reference-implementation/lib/ReadableStreamBYOBRequest-impl.js

```diff
@@ -25,17 +25,14 @@ exports.implementation = class ReadableStreamBYOBRequestImpl {
   }
 
   respondWithNewView(view) {
-    if (view.byteLength === 0) {
-      throw new TypeError('chunk must have non-zero byteLength');
-    }
-    if (view.buffer.byteLength === 0) {
-      throw new TypeError('chunk\'s buffer must have non-zero byteLength');
-    }
-
     if (this._controller === undefined) {
       throw new TypeError('This BYOB request has been invalidated');
     }
 
+    if (CanTransferArrayBuffer(view.buffer) === false) {
```

Comment:
I think it's fine to change this back to `IsDetachedBuffer` too? We'll throw an error on [this line](https://github.com/whatwg/streams/blob/5da8d5c2bfdde4e3ce34ec3e63f4bd3f86922390/reference-implementation/lib/abstract-ops/readable-streams.js#L1339) instead, which is still *before* any state is mutated.